packages/apollo/src/commands/service/__tests__/check.test.ts

```diff
@@ -1,22 +1,328 @@
   let originalChalkEnabled;
 
   beforeEach(() => {
     originalChalkEnabled = chalk.enabled;
     chalk.enabled = false;
+
+    // Clean console log capturing before tests in the event that `afterEach` was not run successfully.
+    uncaptureApplicationOutput();
+
+    // Clean up all network mocks before tests in the event that `afterEach` was not run successfully.
+    nock.cleanAll();
+
+    nock.disableNetConnect();
+
+    // Set the jest timeout to be longer than the default 5000ms to compensate for slow CI.
+    jest.setTimeout(15000);
   });
 
   afterEach(() => {
     chalk.enabled = originalChalkEnabled;
+
+    // Clean up console log mocking
+    uncaptureApplicationOutput();
+
+    // Clean up all network mocks and restore original functionality
+    nock.cleanAll();
+    nock.enableNetConnect();
+  });
+
+  // These are integration tests and not e2e tests because these don't actually hit the remote server.
+  describe("integration", () => {
+    describe("non-federated", () => {
+      describe("should report traffic errors correctly", () => {
+        it("vanilla", async () => {
+          captureApplicationOutput();
+          mockNonFederatedFailure();
+          expect.assertions(2);
+
+          await expect(
+            ServiceCheck.run([...cliKeyParameter])
+          ).rejects.toThrow();
+
+          // Inline snapshots don't work here due to https://github.com/facebook/jest/issues/6744.
+          expect(uncaptureApplicationOutput()).toMatchSnapshot();
+        });
+
+        it("--markdown", async () => {
+          captureApplicationOutput();
+          mockNonFederatedFailure();
+          expect.assertions(2);
+
+          // markdown formatted output should not throw
+          await expect(
+            ServiceCheck.run([...cliKeyParameter, "--markdown"])
+          ).resolves.not.toThrow();
+
+          // Inline snapshots don't work here due to https://github.com/facebook/jest/issues/6744.
+          expect(uncaptureApplicationOutput()).toMatchSnapshot();
+        });
+
+        it("--json", async () => {
+          captureApplicationOutput();
+          mockNonFederatedFailure();
+          expect.assertions(2);
+
+          // JSON formatted output should not throw
+          await expect(
+            ServiceCheck.run([...cliKeyParameter, "--json"])
+          ).resolves.not.toThrow();
+
+          // Inline snapshots don't work here due to https://github.com/facebook/jest/issues/6744.
+          expect(uncaptureApplicationOutput()).toMatchSnapshot();
+        });
+      });
+
+      describe("should report traffic non-errors correctly", () => {
+        it("vanilla", async () => {
+          captureApplicationOutput();
+          mockNonFederatedSuccess();
+          expect.assertions(2);
+
+          await expect(
+            ServiceCheck.run([...cliKeyParameter])
+          ).resolves.not.toThrow();
+
+          // Inline snapshots don't work here due to https://github.com/facebook/jest/issues/6744.
```

Comment:
This issue is left `needs repro` and hasn't seen much activity. Can we give them directions on how to reproduce within our repo after this lands?

Reply:
I can _try_, but I'm not certain _where_ in the `ServiceCheck.run` the issue is coming from. I want to investigate that there's not some weird `fork` or multi-process logic going on.